src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -46,6 +46,13 @@ public LogicManager(Model model, Storage storage) {
     public CommandResult execute(String commandText) throws CommandException, ParseException {
         logger.info("----------------[USER COMMAND][" + commandText + "]");
 
+        ReadOnlyZooKeepBook currentUpdate = new ZooKeepBook();
```

Comment:
This shouldn't be initialized to a new ZKB I think, just declare the variable.

Reply:
I think this variable has to be initialized to some base value, if not the check on line 67 will throw an error saying that it is not initialized yet.